src/indexables-page.php

```diff
@@ -0,0 +1,124 @@
+	 *
+	 * @return int The current post threshold.
+	 */
+	public function indexables_posts_threshold( $number ) {
+		if ( $this->option->get( 'indexables_posts_threshold' ) > 0 ) {
+			return $this->option->get( 'indexables_posts_threshold' );
+		}
+
+		return $number;
+	}
+
+	/**
+	 * Filter minimum analyzed post threshold.
+	 *
+	 * @param int $number The current analyzed post threshold.
+	 *
+	 * @return int The current analyzed post threshold.
+	 */
+	public function indexables_analyzed_posts_threshold( $number ) {
+		if ( $this->option->get( 'indexables_analyzed_posts_threshold' ) > 0 ) {
+			return $this->option->get( 'indexables_analyzed_posts_threshold' );
+		}
+
+		return $number;
+	}
+
+	/**
+	 * Retrieves the controls.
+	 *
+	 * @return string The HTML to use to render the controls.
+	 */
+	public function get_controls() {
+		// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound -- Using WPSEO hook.
+		$placeholder_thresholds = \apply_filters( 'wpseo_posts_threshold', Indexables_Page_Helper::POSTS_THRESHOLD );
+		// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound -- Using WPSEO hook.
+		$placeholder_analyzed_thresholds = \apply_filters( 'wpseo_analyzed_posts_threshold', Indexables_Page_Helper::ANALYSED_POSTS_THRESHOLD );
+
+		$value = '';
+		if ( $this->option->get( 'indexables_posts_threshold' ) > 0 ) {
+			$value = $this->option->get( 'indexables_posts_threshold' );
+		}
+
+		$output  = '<label for="indexables_posts_threshold">' . \esc_html__( 'Minimum threshold for the amount of posts needed:', 'yoast-test-helper' ) . '</label>';
+		$output .= '<input type="number" size="5" value="' . $value . '" placeholder="' . $placeholder_thresholds . '" name="indexables_posts_threshold" id="indexables_posts_threshold"/><br/>';
+
+		$value = '';
+		if ( $this->option->get( 'indexables_analyzed_posts_threshold' ) > 0 ) {
+			$value = $this->option->get( 'indexables_analyzed_posts_threshold' );
+		}
+
+		$output .= '<label for="indexables_analyzed_posts_threshold">' . \esc_html__( 'The minimum threshold for the amount of analyzed posts:', 'yoast-test-helper' ) . '</label>';
+		$output .= '<input type="number" step=".01" size="5" min=0 max=100 value="' . $value . '" placeholder="' . $placeholder_analyzed_thresholds . '" name="indexables_analyzed_posts_threshold" id="indexables_analyzed_posts_threshold"/><span>%</span><br/>';
+
+
+		return Form_Presenter::get_html( \__( 'Integration page thresholds', 'yoast-test-helper' ), 'yoast_seo_test_indexables_page', $output );
+	}
+
+	/**
+	 * Handles the form submit.
+	 *
+	 * @return void
+	 */
+	public function handle_submit() {
+		if ( \check_admin_referer( 'yoast_seo_test_indexables_page' ) !== false ) {
+			$indexables_posts_threshold = null;
+			if ( isset( $_POST['indexables_posts_threshold'] ) ) {
+				$indexables_posts_threshold = \filter_input( \INPUT_POST, 'indexables_posts_threshold', \FILTER_SANITIZE_NUMBER_INT );
```

Comment:
Best if we don't user filter_input here (and below) but rather `sanitize_*` functions.

Similar PR: https://github.com/Yoast/wordpress-seo/pull/18340
Resources behind the reasoning: [PHP8.1 slides - slides #22-23](https://app.slack.com/client/T0258FPMJ/C0258FPMU/files/F034GSLLA4B)